### ISSUE TITLE
Is communication done per-flow, per-device, per-something else?

### ISSUE THREAD

**Issue by mjoras** [mjoras/SCONE-PROTOCL#5]
This came up in the side meeting. The motivating use-case is meant to communicate information that is effectively device-scoped in practice, e.g. the network device may be communicating information that comes from a user's plan information limiting video quality to 2Mbps.

If that information is communicated per flow, then an application could open up numerous connections and then potentially get the  same limit for each, "bypassing" the restrictions.

Does the charter need to address this point about whether a per-flow communication mechanism is the only one to be standardized, and address the potential for "gaming" it?

CC @ihlar

**Comment by SpencerDawkins** [comment]
This is about "gaming the system". We don't need an answer before the BOF, but it would be good to assert that this is a solvable problem at the BOF, if anyone asks

**Comment by ihlar** [comment]
I think it will be difficult to solve gaming in the protocol design. My preferred approach would be to communicate per-flow and let the network device handle any detection and reaction to gaming attempts as it sees fit.

**Comment by SpencerDawkins** [comment]
Tl;dr - I agree with @ihlar on this. I suspect that if we do anything about this, it will be much better for us to put it in the proposed SCONEPRO Protocol Manageability draft proposed in [PR 14](https://github.com/mjoras/SCONE-PROTOCL/pull/14).

I don't think mandating specific reactions is likely to succeed, because reasonable reactions to flow behavior are likely to change over time. We have a lot of experience with proposals in this space that start out describing normative behavior, and end up saying "we're giving you a **hint**. **One thing you might do** when you get this hint **is X**".

If we don't mandate specific reactions, I'm confused about how we could tell people we've prevented gaming.

**Comment by Kevsy** [comment]
Also agree with @ihlar - and to note that per-flow is preferable to per-packet (which would be prohibitively expensive for a network).

**Comment by danwing** [comment]
A nuance is network operators **want** some "gaming" to occur -- or some applications to not use SCONE.  Bandwidth testing applications, for example, need to run 'full speed' so the human feels the network is delivering its promised low latency and high bandwidth.  As SCONE progresses, we should hope bandwidth testing applications also use SCONE and provide a way for the human to test SCONE-throttled flows and full-speed flows.

Without full speed flows, users will feel their existing LTE is not worth upgrading to 5GUW or 6G or 8G or whatever-is-fancy-this-year.

**Comment by Kevsy** [comment]
It would be interesting to see how SCONE impacts user experience in challenging cellular conditions (cell edge, rapid mobility, indoors, cell congestion etc.) .  In these conditions the comparison is not so much SCONE-throttled flows vs  full-speed flows (because you won't achieve 'full speed' vs optimal conditions) but rather SCONE vs non-SCONE.

**Comment by VMatrix1900** [comment]
> A nuance is network operators **want** some "gaming" to occur -- or some applications to not use SCONE. Bandwidth testing applications, for example, need to run 'full speed' so the human feels the network is delivering its promised low latency and high bandwidth. As SCONE progresses, we should hope bandwidth testing applications also use SCONE and provide a way for the human to test SCONE-throttled flows and full-speed flows.
> 
> Without full speed flows, users will feel their existing LTE is not worth upgrading to 5GUW or 6G or 8G or whatever-is-fancy-this-year.

If the speed of the SCONE-throttled flow can be tested, then we do not need SCONE protocol to request it, right? Host can run an iPerf over SCONE-throttled flow and set the maxium rate by its own.

**Comment by jlivingood** [comment]
Is the SCONE signal even useful if it is on a per-device basis? For example, if it signals that ISP X will shape the device's video streaming at 25 Mbps. But the sector of the access network the device is on can only support 10 Mbps at peak hour. So in this case, is the signal even useful? 

And if there are regular times when the signal is not useful, because available bandwidth is less that the signaled rate, why not continue to rely solely upon HTTP adaptive streaming to adjust rate according to available throughput from moment to moment?

**Comment by Kevsy** [comment]
> Is the SCONE signal even useful if it is on a per-device basis? 

I believe it would be on cellular mobile, due to allocated frequency bandwidth (e.g. at cell edge or indoors) and local environmental interference being device-specific.  

> why not continue to rely solely upon HTTP adaptive streaming to adjust rate according to available throughput from moment to moment?

We tested with DASH video and found an improvement using [https://datatracker.ietf.org/doc/html/draft-flinck-mobile-throughput-guidance-04](Mobile Throughput Guidance) under cell congestion / device in poor coverage...as I recall there was little benefit when the device was in an uncontested network with strong signal. Albeit this was 2017 and with TCP.

**Comment by Kevsy** [comment]
I think maximum throughput of a video flow to a given device is the primary use case - and that the device context has to be taken into account for the network (at least, a cellular network) to arrive at that estimate.  Ideally the same mechanism can be used to signal other network conditions/hints and the charter appears to support that.

I think the per-device context is implicit in the current list of properties in the charter. We could add:

 'Associativity with a device: The network properties _may_  be calculated  on a per-device basis, for example in a cellular network where network conditions may vary significantly between devices'  

The other aspect could be any data cap the user has as part of their network subscription (again this is probably more relevant for cellular):

 'Associativity with a user subscription: The network properties _may_  be calculated  on a per-subscriber basis, for example where the subscriber is nearing the limit of a contractual data cap' (NB I'm not sure how that would translate to 'maximum throughput', maybe it should be discussed as another potential signal)  






PS I've raised #77 (with associated PR #78) to change 'maximum achievable bandwidth' to 'maximum achievable throughput'

**Comment by ihlar** [comment]
closed by #84
